### PR TITLE
[scd] oir upsert: factor out determination of subscriptions to notify

### PR DESCRIPTION
**Pull request by Shastick**
Another step towards #1088

### CODE REVIEW

File: pkg/scd/operational_intents_handler.go [hunk 1]
```diff
@@ -508,6 +508,56 @@ func validateUpsertRequestAgainstPreviousOIR(
 	return nil
 }
 
+func incrementIndicesAndGetRelevantSubscriptions(
```
Comment (BenjaminPelletier) [inline pkg/scd/operational_intents_handler.go:511]: nit: I think the primary thing this function is doing is getting relevant subscriptions, with a sub-behavior of incrementing notification indices

```suggestion
func getRelevantSubscriptionsAndIncrementIndices(
```

**Review by BenjaminPelletier — APPROVED** [review]

File: pkg/scd/operational_intents_handler.go [hunk 2]
```diff
@@ -517,6 +517,56 @@ func validateUpsertRequestAgainstPreviousOIR(
 	return nil
 }
 
+func getRelevantSubscriptionsAndIncrementIndices(
```
Comment (mickmis) [inline pkg/scd/operational_intents_handler.go:520]: Drive-by comment: also use this in `DeleteOperationalIntentReference`? There seems to be some common logic (although maybe there are some small deviations)

Comment (Shastick) [reply pkg/scd/operational_intents_handler.go:520]: Good point. I'll quickly check: if the method can be used as-is I'll update to use it, otherwise I'll keep that for another day

Comment (Shastick) [reply pkg/scd/operational_intents_handler.go]: Ended up doing it as it was an easy fix.

**Review by mickmis — APPROVED** [review]
